### PR TITLE
Property Field in ListView

### PR DESCRIPTION
**Pull request by FrancoisGe**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/views/list/list_renderer.js [hunk 1]
```diff
@@ -209,6 +210,37 @@ export class ListRenderer extends Component {
...
+                let nextId = 0;
+                const plop = Object.values(list.activeFields)
+                    .filter(
+                        (activeField) =>
+                            activeField.relatedPropertyField &&
+                            activeField.relatedPropertyField.fieldName === propertiesColumn.name
+                    )
+                    .map((activeField) => ({
+                        ...activeField,
+                        id: `${propertiesColumn.id}_${nextId++}`,
```
Comment (mcm-odoo) [inline addons/web/static/src/views/list/list_renderer.js:228]: maybe set id in the model?

File: addons/web/static/src/views/relational_model.js [hunk 2]
```diff
@@ -1604,6 +1666,92 @@ class DynamicList extends DataPoint {
...
+            propertiesFieldNames.map(async (fieldName) => {
+                const field = this.fields[fieldName];
+                const coModelName = this.fields[field.definition_record].relation;
+                const ids = unique(
+                    records
+                        .map((r) => r[field.definition_record] && r[field.definition_record][0])
+                        .filter(Boolean)
+                );
+                const parentRecords = await this.model.orm.read(coModelName, ids, [
+                    "display_name",
```
Comment (mcm-odoo) [inline addons/web/static/src/views/relational_model.js:1697]: is it needed?

File: addons/web/static/tests/mock_server_tests.js [hunk 3]
```diff
@@ -44,6 +44,12 @@ QUnit.module("MockServer", (hooks) => {
                             relation: "foo",
                             inverse_fname_by_model_name: { foo: "many2many_field" },
                         },
+                        properties: {
```
Comment (mcm-odoo) [inline addons/web/static/tests/mock_server_tests.js:47]: remove diff of this file

File: addons/web/static/tests/views/fields/properties_field_tests.js [hunk 4]
```diff
@@ -21,11 +21,14 @@ async function closePopover(target) {
 }
 
 async function changeType(target, propertyType) {
-    const TYPES_INDEX = {"datetime": 6, "selection": 7, "tags": 8, "many2one": 9, "many2many": 10};
+    const TYPES_INDEX = { datetime: 6, selection: 7, tags: 8, many2one: 9, many2many: 10 };
```
Comment (mcm-odoo) [inline addons/web/static/tests/views/fields/properties_field_tests.js:24]: maybe the diff can be removed?

File: addons/web/static/tests/views/list_view_tests.js [hunk 5]
```diff
@@ -16398,4 +16412,786 @@ QUnit.module("Views", (hooks) => {
...
+            "/web/dataset/call_kw/foo/get_views",
+            "/web/dataset/call_kw/foo/search_read",
+            "/web/dataset/call_kw/bar/read",
+            "/web/dataset/call_kw/foo/web_search_read",
+        ]);
+
+        await toggleFavoriteMenu(target);
+        await toggleMenuItem(target, "only one");
+
+        assert.verifySteps(["/web/dataset/call_kw/foo/web_search_read"]);
```
Comment (mcm-odoo) [inline addons/web/static/tests/views/list_view_tests.js:17156]: wrong, this is duplicated from the test "reload properties definitions when domain change"

File: addons/web/static/tests/views/list_view_tests.js [hunk 6]
```diff
@@ -16398,4 +16412,786 @@ QUnit.module("Views", (hooks) => {
...
+            irFilters: [
+                {
+                    context: "{}",
+                    domain: "[['id', '=', 1]]",
+                    id: 7,
+                    name: "only one",
+                    sort: "[]",
+                    user_id: [2, "Mitchell Admin"],
+                },
+            ],
```
Comment (mcm-odoo) [inline addons/web/static/tests/views/list_view_tests.js:17143]: remove

File: addons/web/static/src/views/kanban/kanban_model.js [hunk 7]
```diff
@@ -372,6 +372,7 @@ export class KanbanDynamicGroupList extends DynamicGroupList {
      * @override
      */
     async load() {
+        await this._loadProperties();
```
Comment (mcm-odoo) [inline addons/web/static/src/views/kanban/kanban_model.js:375]: check if we can remove

File: addons/web/static/src/views/fields/tags/tags_field.js [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+import { AutoComplete } from "@web/core/autocomplete/autocomplete";
+import { sprintf } from "@web/core/utils/strings";
+import { Component } from "@odoo/owl";
+
+class PropertyTagsColorListPopover extends Component {}
+PropertyTagsColorListPopover.template = "web.PropertyTagsColorListPopover";
+PropertyTagsColorListPopover.components = {
+    ColorList,
+};
+
```
Comment (FrancoisGe) [inline addons/web/static/src/views/fields/tags/tags_field.js:19]: add static

File: addons/web/static/src/views/fields/tags/tags_field.js [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+// property tags does not really need timeout because it does not make RPC calls
+export class PropertyTagAutoComplete extends AutoComplete {}
+Object.assign(PropertyTagAutoComplete, { timeout: 0 });
```
Comment (FrancoisGe) [inline addons/web/static/src/views/fields/tags/tags_field.js:22]: add prop timeout

File: addons/web/static/src/views/list/list_renderer.js [hunk 10]
```diff
@@ -209,6 +209,36 @@ export class ListRenderer extends Component {
...
+                        classNames: propertiesColumn.classNames,
+                        optional: "hide",
+                        type: "field",
+                        hasLabel: true,
+                        label: activeField.string,
+                    }));
+            } else if (propertiesColumn.relatedPropertyField) {
+                return [];
```
Comment (FrancoisGe) [inline addons/web/static/src/views/list/list_renderer.js:235]: ??? remove

File: addons/web/static/src/views/list/list_renderer.js (hunk 1 above)
Comment (FrancoisGe) [reply addons/web/static/src/views/list/list_renderer.js:227]: use juste name prop

File: addons/web/static/src/views/list/list_renderer.js [hunk 11]
```diff
@@ -464,16 +498,34 @@ export class ListRenderer extends Component {
...
+            };
+            if (result[id]) {
+                result[id].optionalFields.push(optionalField);
+            } else {
+                result[id] = { id, displayName, optionalFields: [optionalField] };
+            }
+        }
+        return Object.values(result);
+    }
+
```
Comment (FrancoisGe) [inline addons/web/static/src/views/list/list_renderer.js:526]: Has one getter with group utils for studio ??? POC in other commit

File: addons/web/static/src/views/list/list_renderer.xml [hunk 12]
```diff
@@ -222,7 +239,15 @@
...
-                    <td t-on-keydown.synthetic="(ev) => this.onCellKeydown(ev, group, record)"
+                    <t t-if="column.relatedPropertyField and !(column.name in record.data)">
+                        <td t-on-keydown.synthetic="(ev) => this.onCellKeydown(ev, group, record)"
+                            class="o_data_cell cursor-pointer bg-light"
+                            t-att-name="column.name"
+                            t-att-colspan="column.colspan"
+                            t-on-click="(ev) => this.onCellClicked(record, column, ev)"
+                            tabindex="-1" />
+                    </t>
+                    <td t-else="" t-on-keydown.synthetic="(ev) => this.onCellKeydown(ev, group, record)"
```
Comment (FrancoisGe) [inline addons/web/static/src/views/list/list_renderer.xml:250]: try to merge

File: addons/web/static/src/views/relational_model.js [hunk 13]
```diff
@@ -416,6 +429,8 @@ export class Record extends DataPoint {
         this._changes = {};
         this.data = {};
 
+        this.fieldsProperties = {};
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model.js:432]: remove

File: addons/web/static/src/views/relational_model.js [hunk 14]
```diff
@@ -1604,6 +1680,92 @@ class DynamicList extends DataPoint {
     // Protected
     // -------------------------------------------------------------------------
 
+    async _loadProperties() {
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model.js:1683]: move to model

File: addons/web/static/src/views/relational_model.js [hunk 15]
```diff
@@ -1604,6 +1680,92 @@ class DynamicList extends DataPoint {
...
+        if (this.propertiesLoaded) {
+            return;
+        }
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model.js:1686]: to remove

File: addons/web/static/src/views/relational_model.js [hunk 16]
```diff
@@ -1604,6 +1680,92 @@ class DynamicList extends DataPoint {
...
+
+        const propertiesFieldNames = this.fieldNames.filter(
+            (name) => this.fields[name].type === "properties"
+        );
+        if (!propertiesFieldNames.length) {
+            return;
+        }
+
+        const records = await this.model.orm.searchRead(
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model.js:1695]: search

**Pull request closed without merging**